design/cdk-bootstrap.md

```diff
@@ -0,0 +1,444 @@
+      }
+    },
+    "AssumedActionRole": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {
+                  "Ref": "TrustedAccounts"
+                }
+              }
+            }
+          ]
+        },
+        "Policies": [
+          {
+            "PolicyDocument": {
+              "Statement": [
+                {
+                  "Action": [
+                    "cloudformation:CreateChangeSet",   "cloudformation:DeleteChangeSet",
+                    "cloudformation:DescribeChangeSet", "cloudformation:DescribeStacks",
+                    "cloudformation:ExecuteChangeSet",
+                    "s3:GetObject*",                    "s3:GetBucket*",
+                    "s3:List*",                         "s3:Abort*",
+                    "s3:DeleteObject*",                 "s3:PutObject*",
+                    "kms:Decrypt",                      "kms:DescribeKey"
+                  ],
+                  "Resource": "*"
+                },
+                {
+                  "Action": "iam:PassRole",
+                  "Resource": {
+                    "Fn::Sub": "${CfnDeploymentRole.Arn}"
+                  }
+                }
+              ]
+            },
+            "PolicyName": {
+              "Fn::Sub": "DefaultBootstrapAssumedActionRolePolicy-${AWS::Region}"
+            }
+          }
+        ],
+        "Condition": "HasTrustedAccounts"
+      }
+    },
+    "CfnDeploymentRole": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Effect": "Allow",
+              "Principal": {
+                "Service": "cloudformation.amazonaws.com"
+              }
+            }
+          ]
+        },
+        "ManagedPolicyArns": {
+          "Ref": "CfnDeploymentRoleManagedPolicies"
+        },
+        "Condition": "HasTrustedAccounts"
+      }
+    }
+  },
+  "Outputs": {
+    "BucketName": {
+      "Description": "The name of the S3 bucket owned by the CDK toolkit stack",
+      "Value": { "Fn::Sub":  "${StagingBucket.Arn}" },
+      "Export": {
+        "Name": { "Fn::Sub": "${AWS::StackName}:BucketName" }
+      }
+    },
+    "BucketDomainName": {
+      "Description": "The domain name of the S3 bucket owned by the CDK toolkit stack",
+      "Value": { "Fn::Sub":  "${StagingBucket.RegionalDomainName}" },
+      "Export": {
+        "Name": { "Fn::Sub": "${AWS::StackName}:BucketDomainName" }
+      }
+    },
+    "BootstrapVersion": {
+      "Description": "The version of the bootstrap resources that are currently mastered in this stack",
+      "Value": "1",
+      "Export": {
+        "Name": { "Fn::Sub": "AwsCdkBootstrapVersion-${AWS::Region}" }
```

Comment:
why is the region needed here?
The prefix should be included in the export name

I still don’t understand who this value is going to be used.

Reply:
Removed `${AWS::Region}`.